runtime/autoload/spellfile.vim

```diff
@@ -196,9 +196,11 @@ function! spellfile#GetDirChoices()
 endfunc
 
 function! spellfile#WritableSpellDir()
-  if has("unix")
-    " For Unix always use the $HOME/.nvim directory
-    return $HOME . "/.nvim/spell"
+  " Always use the $XDG_DATA_HOME/nvim/site directory
+  if exists('$XDG_DATA_HOME')
+    return $XDG_DATA_HOME . "/nvim/site/spell"
+  else
+    return $HOME . "/.local/share/nvim/site/spell"
```

Comment:
@justinmk The remaining body is neither correct. One should not iterate over `split(&rtp, ',')`, there is `globpath()`. If `globpath()` is not to be used, then code must do unescaping and split on rather lengthy regex (you may see examples in VAM and pathogen code, also in many others).